src/main/scala/akka/monitor/instrumentation/DispatcherInstrumentation.scala

```diff
@@ -83,9 +83,10 @@ class DispatcherInstrumentation {
       executorService match {
         case tpe: ThreadPoolExecutor => ExecutorServiceMetrics.monitor(AkkaMetricRegistry.getRegistry, tpe, prefixedName, Tag.of("type", "ThreadPoolExecutor"))
         case fjp: ForkJoinPool => ExecutorServiceMetrics.monitor(AkkaMetricRegistry.getRegistry, fjp, prefixedName, Tag.of("type", "ForkJoinPool"))
-        case afjp: akka.dispatch.forkjoin.ForkJoinPool =>
-          ExecutorServiceMetrics.monitor(AkkaMetricRegistry.getRegistry, afjp, prefixedName, Tag.of("type", "ForkJoinPool"))
-        case es: ExecutorService => ExecutorServiceMetrics.monitor(AkkaMetricRegistry.getRegistry, es, prefixedName, Tag.of("type", es.getClass.getName))
+        case fjp: akka.dispatch.ForkJoinExecutorConfigurator.AkkaForkJoinPool =>
+          ExecutorServiceMetrics.monitor(AkkaMetricRegistry.getRegistry, fjp, prefixedName, Tag.of("type", "ForkJoinPool"))
```

Comment:
This will not work and broke the functionality I am sure. Please take a look at the implementation of `ExecutorServiceMetrics`. They also do an instanceof match and that one will never match  `akka.dispatch.ForkJoinExecutorConfigurator.AkkaForkJoinPool`. I think the original approach you take is the right one and should not be changed and we also should not rely on `ExecutorServiceMetrics`. I tried very hard to do that in our code and failed.

I can provide a PR also later

Reply:
https://github.com/pjfanning/micrometer-akka-sample/tree/executor-service-style -- this branch uses the snapshot release and is configured to use the new style metrics.

It reports metrics like
```
# HELP executor_completed_tasks_total The approximate total number of tasks that have completed execution
# TYPE executor_completed_tasks_total counter
executor_completed_tasks_total{name="akka.io.pinned-dispatcher",type="ThreadPoolExecutor",} 0.0
executor_completed_tasks_total{name="akka.stream.default-blocking-io-dispatcher",type="ThreadPoolExecutor",} 9.0
# HELP executor_idle_seconds  
# TYPE executor_idle_seconds summary
executor_idle_seconds_count{name="WebServer_akka.actor.default-dispatcher",type="ForkJoinPool",} 0.0
executor_idle_seconds_sum{name="WebServer_akka.actor.default-dispatcher",type="ForkJoinPool",} 0.0
executor_idle_seconds_count{name="akka.io.pinned-dispatcher",type="ThreadPoolExecutor",} 0.0
executor_idle_seconds_sum{name="akka.io.pinned-dispatcher",type="ThreadPoolExecutor",} 0.0
executor_idle_seconds_count{name="akka.stream.default-blocking-io-dispatcher",type="ThreadPoolExecutor",} 0.0
executor_idle_seconds_sum{name="akka.stream.default-blocking-io-dispatcher",type="ThreadPoolExecutor",} 0.0
# HELP executor_idle_seconds_max  
# TYPE executor_idle_seconds_max gauge
executor_idle_seconds_max{name="WebServer_akka.actor.default-dispatcher",type="ForkJoinPool",} 0.0
executor_idle_seconds_max{name="akka.io.pinned-dispatcher",type="ThreadPoolExecutor",} 0.0
executor_idle_seconds_max{name="akka.stream.default-blocking-io-dispatcher",type="ThreadPoolExecutor",} 0.0
# HELP akka_system_actor_count  
# TYPE akka_system_actor_count gauge
akka_system_actor_count{actorSystem="WebServer",} 18.0
# HELP executor_queued_tasks The approximate number of tasks that are queued for execution
# TYPE executor_queued_tasks gauge
executor_queued_tasks{name="akka.io.pinned-dispatcher",type="ThreadPoolExecutor",} 0.0
executor_queued_tasks{name="akka.stream.default-blocking-io-dispatcher",type="ThreadPoolExecutor",} NaN
# HELP executor_pool_size_threads The current number of threads in the pool
# TYPE executor_pool_size_threads gauge
executor_pool_size_threads{name="akka.io.pinned-dispatcher",type="ThreadPoolExecutor",} 1.0
executor_pool_size_threads{name="akka.stream.default-blocking-io-dispatcher",type="ThreadPoolExecutor",} NaN
# HELP executor_queue_remaining_tasks The number of additional elements that this queue can ideally accept without blocking
# TYPE executor_queue_remaining_tasks gauge
executor_queue_remaining_tasks{name="akka.io.pinned-dispatcher",type="ThreadPoolExecutor",} 2.147483647E9
executor_queue_remaining_tasks{name="akka.stream.default-blocking-io-dispatcher",type="ThreadPoolExecutor",} NaN
# HELP akka_router_message_count_webserver_system_io_tcp_selectors_total  
# TYPE akka_router_message_count_webserver_system_io_tcp_selectors_total counter
akka_router_message_count_webserver_system_io_tcp_selectors_total 9.0
# HELP executor_seconds  
# TYPE executor_seconds summary
executor_seconds_count{name="WebServer_akka.actor.default-dispatcher",type="ForkJoinPool",} 0.0
executor_seconds_sum{name="WebServer_akka.actor.default-dispatcher",type="ForkJoinPool",} 0.0
executor_seconds_count{name="akka.io.pinned-dispatcher",type="ThreadPoolExecutor",} 0.0
executor_seconds_sum{name="akka.io.pinned-dispatcher",type="ThreadPoolExecutor",} 0.0
executor_seconds_count{name="akka.stream.default-blocking-io-dispatcher",type="ThreadPoolExecutor",} 0.0
executor_seconds_sum{name="akka.stream.default-blocking-io-dispatcher",type="ThreadPoolExecutor",} 0.0
# HELP executor_seconds_max  
# TYPE executor_seconds_max gauge
executor_seconds_max{name="WebServer_akka.actor.default-dispatcher",type="ForkJoinPool",} 0.0
executor_seconds_max{name="akka.io.pinned-dispatcher",type="ThreadPoolExecutor",} 0.0
executor_seconds_max{name="akka.stream.default-blocking-io-dispatcher",type="ThreadPoolExecutor",} 0.0
# HELP akka_actor_group_actor_count  
# TYPE akka_actor_group_actor_count gauge
akka_actor_group_actor_count{groupName="all",} 3.0
# HELP executor_active_threads The approximate number of threads that are actively executing tasks
# TYPE executor_active_threads gauge
executor_active_threads{name="akka.io.pinned-dispatcher",type="ThreadPoolExecutor",} 1.0
executor_active_threads{name="akka.stream.default-blocking-io-dispatcher",type="ThreadPoolExecutor",} NaN
```

Reply:
Ok, now I am lost. I have no idea how this could work :)
Ok, I will try the new release later